### PR TITLE
Move plugin JS to main browser code

### PR DESCRIPTION
**Pull request by fortuna**
The plugin JS is an independent compile unit that is not actually compiled, so it can't depend on other libraries or shared easily.

This moves all the JS logic into the main browser code as Typescript, so it's easier to understand the logic, reuse code and modify the interfaces as we move to unify Cordova and Electron.

@alalamav Would you mind sanity checking my PR?

/cc @daniellacosse @bemasc FYI

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,8 +11,10 @@
     "reset": "npm run clean && npm ci"
   },
   "comments": {
-    "sentry": "@sentry/electron depends on @sentry/browser; use @sentry/browser as a transitive dependency to avoid version mismatches that make the build fail. See: https://github.com/getsentry/sentry-javascript/issues/1853",
-    "codrova-osx@6.0.0": "Version-controlled platform config files at apple/xcode/osx/Outline/config.xml, apple/xcode/osx/osx.json, and apple/xcode/osx/www/cordova_plugins.js as a workaround for https://github.com/apache/cordova-osx/issues/106. Delete these files when the issue is fixed."
+    "sentry":
```
Comment (daniellacosse) [inline package.json:14]: Can we do something about this? Every time we do npm it formats this back. Perhaps we can ignore `package.json` in the linter

File: src/www/app/cordova_main.ts [hunk 2]
```diff
@@ -49,22 +55,69 @@ class CordovaClipboard extends AbstractClipboard {
...
+}
+
 // Adds reports from the (native) Cordova plugin.
-export class CordovaErrorReporter extends SentryErrorReporter {
+class CordovaErrorReporter extends SentryErrorReporter {
   constructor(appVersion: string, appBuildNumber: string, dsn: string) {
     super(appVersion, dsn, {'build.number': appBuildNumber});
-    cordova.plugins.outline.log.initialize(dsn).catch(console.error);
+    // Initializes the error reporting framework with the supplied credentials.
+    // TODO(fortuna): This is an Promise that is not waited for and can cause a race condition.
```
Comment (daniellacosse) [inline src/www/app/cordova_main.ts:78]: Ah, I thought we didn't use `TODO(username)` in our CB, good to know

File: package.json (hunk 1 above)
Comment (fortuna) [reply package.json]: Done, I think, with `.clang-format-ignore`

**Review by fortuna — COMMENTED** [review]
App is working fine on macOS.

**Review by daniellacosse — APPROVED** [review]

**Review by alalamav — APPROVED** [review]
nice change, I like how you integrated the outline plugin change into the browser build. one thing to note is that this change splits the outline plugin in a way that cannot be reused accross cordova projects. maybe that's fine, since the plugin does depend on third party dependencies committed outside the plugin source, and is unlikely being used as as standalone plugin.

**Review by jyyi1 — APPROVED** [review]

File: apple/xcode/osx/osx.json [hunk 3]
```diff
@@ -43,14 +43,6 @@
         "device"
       ]
     },
-    {
```
[the remark this answers is not in the document]
Comment (fortuna) [reply apple/xcode/osx/osx.json:46]: We don't. I tested and the app worked fine. I inspected the project and the dependencies are all listed there. That's what defines what gets linked and it's taken from the config xml. This file is really just for the JS code.

**Comment by fortuna** [comment]
@alalamav I don't see our plugin as a reusable piece of code. That's our app code, so I'm not worried about that.
I do consider extracting parts of it into standalone native libraries that can be developed without the nightmare of shufling files and changes back and forth.

**Pull request merged**